src/preact/base-element.js

```diff
@@ -1045,24 +1062,30 @@ function collectProps(Ctor, element, ref, defaultProps, mediaQueryProps) {
     }
   }
 
-  if (mediaQueryProps) {
-    mediaQueryProps.complete();
-  }
-
-  return props;
-}
-
-/**
- * @param {!Object} props
- * @param {!Object} propDefs
- * @param {!Element} element
- * @param {?MediaQueryProps} mediaQueryProps
- */
-function parsePropDefs(props, propDefs, element, mediaQueryProps) {
   for (const name in propDefs) {
     const def = /** @type {!AmpElementPropDef} */ (propDefs[name]);
+    devAssert(
+      !!def.attr +
+        !!def.attrs +
+        !!def.attrPrefix +
+        !!def.selector +
+        !!def.passthrough +
+        !!def.passthroughNonEmpty <=
+        1,
+      ONE_OF_ERROR_MESSAGE
+    );
     let value;
-    if (def.attr) {
+    if (def.passthrough) {
+      devAssert(Ctor['usesShadowDom']);
+      props[def.name || name] = [<Slot />];
+      continue;
```

Comment:
Nit. But I think `continue` is much clearer w/o `else if`.

Reply:
Removed `continue` instead :)